src/components/common/ServerErrorComponent.tsx

```
@@ -2,14 +2,17 @@ import { RefreshCcw } from "lucide-react"
+    onRetry: (...args: any[]) => void
 }
 
 export default function ServerErrorComponent({ message, onRetry }: ServerErrorComponentProps) {
     return (
         <div className="flex flex-col items-center justify-center gap-2 px-10 py-5 rounded-md">
             <p className="text-gray-500">{message}</p>
-            <button className="bg-blue-500 text-white px-4 py-2 rounded-md flex items-center gap-2" onClick={onRetry}>
+            <button 
+                className="bg-blue-500 text-white px-4 py-2 rounded-md flex items-center gap-2" 
+                onClick={() => onRetry()}
+            >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)